src/language/tsLanguageService.ts

```diff
@@ -0,0 +1,215 @@
+      } else */
+
+      if (name === `${MDB_CONFIG_LIBRARY_NAME}.d.ts`) {
+        libPath = join(this._extensionPath, 'node_modules/mongodb', name);
+      } else if (name === `${GLOBAL_CONFIG_LIBRARY_NAME}.d.ts`) {
+        libPath = join(this._extensionPath, 'src/types', name);
+      } else {
+        content = '';
+      }
+
+      if (libPath) {
+        try {
+          content = readFileSync(libPath, 'utf8');
+        } catch (e) {
+          this._connection.console.error(
+            `Unable to load library ${name} at ${libPath}`
+          );
+          content = '';
+        }
+      } else {
+        content = '';
+      }
+
+      contents[name] = content;
+    }
+
+    return content;
+  }
+
+  _getTypeScriptServiceHost() {
+    const compilerOptions = {
+      allowNonTsExtensions: true,
+      allowJs: true,
+      lib: [
+        `${MDB_CONFIG_LIBRARY_NAME}.d.ts`,
+        `${GLOBAL_CONFIG_LIBRARY_NAME}.d.ts`,
+      ], // , `lib.${TS_CONFIG_LIBRARY_NAME}.d.ts`],
+      target: ts.ScriptTarget.Latest,
+      moduleResolution: ts.ModuleResolutionKind.Classic,
+      experimentalDecorators: false,
+    };
+    let currentTextDocument = TextDocument.create('init', 'javascript', 1, '');
+
+    // Create the language service host to allow the LS to communicate with the host.
+    const host: ts.LanguageServiceHost = {
+      getCompilationSettings: () => compilerOptions,
+      getScriptFileNames: () => [currentTextDocument.uri],
+      getScriptKind: () => {
+        return ts.ScriptKind.JS;
+      },
+      getScriptVersion: (fileName: string) => {
+        if (fileName === currentTextDocument.uri) {
+          return String(currentTextDocument.version);
+        }
+        return '1';
+      },
+      getScriptSnapshot: (fileName: string) => {
+        let text = '';
+        if (fileName === currentTextDocument.uri) {
+          text = currentTextDocument.getText();
+        } else {
+          text = this._loadLibrary(fileName);
+        }
+        return {
+          getText: (start, end) => text.substring(start, end),
+          getLength: () => text.length,
+          getChangeRange: () => undefined,
+        };
+      },
+      getCurrentDirectory: () => '',
+      getDefaultLibFileName: () => GLOBAL_CONFIG_LIBRARY_NAME,
+      readFile: (path: string): string | undefined => {
+        if (path === currentTextDocument.uri) {
+          return currentTextDocument.getText();
+        }
+        return this._loadLibrary(path);
+      },
+      fileExists: (): boolean => false,
+      directoryExists: (): boolean => false,
+    };
+
+    // Create the language service files.
+    const jsLanguageService = ts.createLanguageService(host);
+
+    return {
+      // Return a new instance of the language service.
+      getLanguageService(jsDocument: TextDocument): ts.LanguageService {
+        currentTextDocument = jsDocument;
+        return jsLanguageService;
+      },
```

Comment:
The comment above this function and its name are a bit misleading – the function always returns the same instance, not a new one. Would it make more sense to either a) create a new language service instance each time or b) split this function into two functions, `getLanguageService(): ts.LanguageService` and `setCurrentTextDocument(jsDocument: TextDocument): void`?